### PR TITLE
Search notes for search terms in any order.

### PR DESCRIPTION
**Pull request by jeff-h**
Improve search leniency as per https://github.com/glushchenko/fsnotes/issues/27

### CODE REVIEW

File: FSNotes/ViewController.swift [hunk 1]
```diff
@@ -200,19 +200,18 @@ class ViewController: NSViewController,
...
                         $0.isRemoved == false
                         && (
                             filter.isEmpty
                             || (
-                                !filter.isEmpty
-                                && (
-                                    $0.content.localizedCaseInsensitiveContains(filter)
-                                    || $0.name.localizedCaseInsensitiveContains(filter)
-                                )
+                                searchContent.localizedCaseInsensitiveContainsTerms(searchTermsArray)
```
Comment (glushchenko) [inline FSNotes/ViewController.swift]: Error here "Value of type 'String' has no member 'localizedCaseInsensitiveContainsTerms'"

Comment (jeff-h) [reply FSNotes/ViewController.swift]: Ah, I didn't commit the project/workspace file, sorry! You will need to drag `String+.swift` from `/Utils` into the project sidebar in Xcode, I guess?

Comment (glushchenko) [reply FSNotes/ViewController.swift]: Yep, but then did not work too. Error in String+ "Cannot call value of non-function type 'Substring?'"

Comment (jeff-h) [reply FSNotes/ViewController.swift]: Grrr — which version of Xcode are you on?

Comment (glushchenko) [reply FSNotes/ViewController.swift]: Version 9.0 beta 3 (9M174d)

Comment (glushchenko) [reply FSNotes/ViewController.swift]: Can you try to clean checkout pull request and build again?

Comment (jeff-h) [reply FSNotes/ViewController.swift]: ok

Comment (jeff-h) [reply FSNotes/ViewController.swift]: (Also, I am on Xcode 9.0 beta 5 — is it possibly related to that difference?)

Comment (glushchenko) [reply FSNotes/ViewController.swift]: May be. I will try update xcode today.

Comment (jeff-h) [reply FSNotes/ViewController.swift]: I can't replicate the problem you're seeing, so I'm hoping it's due to the version of Xcode / Swift. I checked and I am using Swift 3.2. Could you see what yours says? It's under **Build Settings > Swift Language Version"**

Comment (glushchenko) [reply FSNotes/ViewController.swift]: Working fine on Xcode 9 beta 6. Thanks @jeff-h

**Pull request merged**